client/main.cpp

```diff
@@ -0,0 +1,176 @@
+*  editing and translation, are reserved. The use of this project is subject to the terms of a
+*  license agreement to be concluded with nymea GmbH in accordance with the terms
+*  of use of nymea GmbH, available under https://nymea.io/license
+*
+*  GNU General Public License Usage
+*  Alternatively, this project may be redistributed and/or modified under
+*  the terms of the GNU General Public License as published by the Free Software Foundation,
+*  GNU version 3. this project is distributed in the hope that it will be useful, but WITHOUT ANY
+*  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+*  PURPOSE. See the GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License along with this project.
+*  If not, see <https://www.gnu.org/licenses/>.
+*
+*  For any further details and any questions please contact us under contact@nymea.io
+*  or see our FAQ/Licensing Information on https://nymea.io/license/faq
+*
+* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#include "mqttclient.h"
+
+#include <QCoreApplication>
+#include <QCommandLineParser>
+#include <QUuid>
+#include <QHostAddress>
+
+
+int main(int argc, char *argv[]) {
+
+    QCoreApplication *app = new QCoreApplication(argc, argv);
+
+    QCommandLineParser parser;
+    parser.addPositionalArgument("server", "The server address.");
+    parser.addOptions({
+          {{"clientid", "c"}, "The client ID to use for the connection (default: autogenerated).", "clientId", QUuid::createUuid().toString()},
+          {{"username", "u"}, "The user name to use for the connection.", "username", ""},
+          {{"password", "P"}, "The password to use for the connection.", "password", ""},
+          {{"subscribe", "s"}, "Subscribe to a topic filter.", "topicfilter"},
+          {{"publish", "p"}, "Publish to topic.", "topic"},
+          {{"payload", "l"}, "Publish payload (requires -p).", "payload"},
+          {{"retain", "r"}, "Retain flag for publishes (default: no retaining)."},
+          {{"qos", "q"}, "QoS (default: 1).", "QoS", "1"},
+          {{"ssl", "S"}, "Use SSL for TCP connections (default: off) (Websocket connections will determine the use of SSL using the url scheme)."},
+          {{"accept-self-signed-certificate", "A"}, "Ignore self signed certificate errors"},
+          {{"verbose", "v"}, "Be more verbose"}
+      });
+    parser.addHelpOption();
+
+    parser.process(app->arguments());
+
+    if (parser.positionalArguments().count() < 1) {
+        parser.showHelp(-1);
+    }
+
+    QString verbose = parser.isSet("verbose") ? "true" : "false";
+    QLoggingCategory::setFilterRules(QString("nymea.mqtt.client.debug=%1\nnymea.mqtt.client.warning=%2\n").arg(verbose).arg(verbose));
+
+    bool ok;
+    Mqtt::QoS qos = static_cast<Mqtt::QoS>(parser.value("qos").toInt(&ok));
```

Comment:
you verify the convert, but not the range

Reply:
fixed